### PR TITLE
fix missing URL formatting

### PR DESCRIPTION
**Pull request by amir-arad**
make `normalizeEndpointForServer` encode endpoint as legal URL

### CODE REVIEW

**Comment by amir-arad** [comment]
a lot of noise in the branch history, worth squashing when merging

File: scripts/src/components.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ import * as descriptors from "./common/descriptors";
 
 import { ConnectorBuilder } from "./connector";
 import * as server from "./server";
+import "./encodeurl";
+import encodeUrl = require('encodeurl')
```
Comment (nitzantomer) [inline scripts/src/components.ts]: Please use double quotes and end with a semicolon.
Yeah, I know, we need lint rules.

File: scripts/src/components.ts [hunk 2]
```diff
@@ -190,8 +192,7 @@ function normalizeEndpointForServer(endpoint: string): string {
 	while ((match = re.exec(endpoint)) !== null) {
 		endpoint = endpoint.replace(match[0], ":" + match[1]);
 	}
-
-	return endpoint;
+	return encodeUrl(endpoint);
```
Comment (nitzantomer) [inline scripts/src/components.ts:195]: just wondering, what was the use case that triggered this change?

**Review by nitzantomer — CHANGES_REQUESTED** [review]

File: scripts/src/components.ts (hunk 2 above)
Comment (amir-arad) [reply scripts/src/components.ts:195]: hebrew commands...

**Comment by amir-arad** [comment]
please publish `@fugazi/connector` to npm when merged

**Review by nitzantomer — APPROVED** [review]

**Pull request merged**